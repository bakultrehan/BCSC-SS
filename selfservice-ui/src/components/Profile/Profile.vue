/** * Complete Profile of app */

<template>
  <v-card class="mx-auto card-width">
    <v-toolbar flat class="bc-subtitle" dark>
      <v-toolbar-title>{{ $t('profile.pagetitle') }}</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-form ref="form" v-model="form">
      <v-container>
        <v-row class="ma-5">
          <v-col cols="12" md="12">
            <v-card-subtitle
              class="text-left padding-0 bc-padding-0"
              v-html="$t('profile.titlePageInfo')"
            ></v-card-subtitle>

            <v-card-title class="text-left bc-padding-0">
              {{ userProfile.firstName }}
              {{ userProfile.lastName }}</v-card-title
            >
            <Input
              v-model="email"
              :label="$t('profile.labelEmail')"
              type="text"
              :rules="[
                rules.required,
                rules.email,
                rules.length(2),
                rules.maxLength(250)
              ]"
              v-if="filedsToShow.email"
            />
            <Input
              v-model="phone"
              :label="$t('profile.labelPhone')"
              type="text"
              :rules="[rules.required, rules.length(9), rules.maxLength(15)]"
              v-if="filedsToShow.phone"
            />
          </v-col>
          <v-col cols="12">
            <v-card flat>
              <v-divider></v-divider>
              <v-card-actions>
                <v-spacer></v-spacer>
                <Button
                  :disabled="!form"
                  class="white--text"
                  color="indigo accent-4"
                  depressed
                  @click="completeProfile"
                  @keyup.enter="completeProfile"
                  >{{ $t('profile.btnContinue') }}</Button
                >
              </v-card-actions>
            </v-card>
          </v-col>
        </v-row>
      </v-container>
    </v-form>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Input from '@/Atomic/Input/Input.vue';
import Button from '@/Atomic/Button/Button.vue';
import validationRules from '@/config/validationRules';

const KeyCloakModule = namespace('KeyCloakModule');

@Component({ components: { Input, Button } })
export default class Dashboard extends Vue {
  @KeyCloakModule.Getter('filedsToShow')
  private filedsToShow!: any;
  @KeyCloakModule.Action('updateProfile')
  private updateProfile!: any;
  @KeyCloakModule.Getter('userProfile') private userProfile!: [];
  @KeyCloakModule.Action('errorStatus')
  private errorStatus!: any;

  private form: boolean = false;

  /* istanbul ignore next */
  private rules = validationRules;
  private email: string = '';
  private phone: string = '';

  private completeProfile() {
    const profile = { email: this.email, phone: this.phone };
    this.updateProfile(profile);
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.card-width {
  max-width: 80%;
  @include lg {
    max-width: 50%;
  }
}
</style>
