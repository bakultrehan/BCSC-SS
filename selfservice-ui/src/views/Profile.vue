<template>
  <div class="home">
    <Profile />
  </div>
</template>

<script lang="ts">
// @ is an alias to /src
import { Component, Vue, Prop } from 'vue-property-decorator';
import Profile from '@/components/Profile/Profile.vue';

@Component({
  components: {
    Profile
  }
})
export default class ProfileHome extends Vue {}
</script>
