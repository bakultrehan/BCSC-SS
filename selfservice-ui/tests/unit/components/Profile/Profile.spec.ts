import { shallowMount, mount } from '@vue/test-utils';
import Profile from '@/components/Profile/Profile.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import Vue from 'vue';
let vuetify: any;
vuetify = new Vuetify();

describe('Profile.vue', () => {
  const store = new Vuex.Store({
    modules: {
      KeyCloakModule: {
        namespaced: true,
        state: {},
        getters: {
          filedsToShow: jest.fn(() => {
            return { email: true, phone: true };
          }),
          userProfile: jest.fn(() => {
            return {
              email: 'asd1@gov.bc.ca',
              firstName: 'John',
              id: 28,
              lastName: 'Doe',
              phone: '0987654321',
            };
          }),
          errorStatus: jest.fn(() => {
            return false;
          }),
          profileErrorStatus: jest.fn(),
          emailExistErrorStatus: jest.fn(),
        },
        actions: {
          updateProfile: jest.fn(),
          errorStatus: jest.fn(),
        },
      },
    },
  });

  const mountFunction = (options: any) => {
    return mount(Profile, {
      store,
      vuetify,
      sync: false,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options,
    });
  };

  it('renders props when passed with gettors', () => {
    const profilePage = mountFunction({});
    expect(profilePage.element).toMatchSnapshot();
  });

  it('renders popup on button click', async () => {
    const profilePage = mountFunction({});
    const toggleDisclaimer = jest.fn();
    profilePage.setData({
      phone: '0987654321',
      email: 'asd2@gov.bc.ca',
    });
    await Vue.nextTick();
    profilePage.setData({ valid: true });
    await Vue.nextTick();
    const button = profilePage.find('.btn-profile-update');
    profilePage.vm.$on('action-btn:clicked', toggleDisclaimer);
    button.trigger('click');
    expect(profilePage.element).toMatchSnapshot();
  });

  it('renders props when passed with gettors', async () => {
    const profilePage = mountFunction({});
    const toggleDisclaimer = jest.fn();
    const createOrUpdateProfile = jest.fn();
    profilePage.setData({
      phone: '0987654321',
      email: 'asd2@gov.bc.ca',
    });
    await Vue.nextTick();
    profilePage.setData({ valid: true });
    await Vue.nextTick();
    const button = profilePage.find('.btn-profile-update');
    profilePage.vm.$on('action-btn:clicked', toggleDisclaimer);
    button.trigger('click');
    await Vue.nextTick();
    profilePage.setData({ buttonEnable: true });
    await Vue.nextTick();
    const button2 = profilePage.find('.btn-submit-terms-profile');
    profilePage.vm.$on('action-btn:clicked', createOrUpdateProfile);
    button2.trigger('click');

    expect(profilePage.element).toMatchSnapshot();
  });
});
