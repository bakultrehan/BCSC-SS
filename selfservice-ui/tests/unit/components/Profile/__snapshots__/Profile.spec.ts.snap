// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.vue renders popup on button click 1`] = `
<div
  class="mx-auto card-width v-card v-sheet theme--light"
>
  <!---->
   
  <header
    class="bc-subtitle v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <div
        class="v-toolbar__title"
      >
        
      </div>
       
      <div
        class="flex-grow-1"
      />
    </div>
  </header>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
   
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <div
      class="container"
    >
      <div
        class="row ma-5"
      >
        <div
          class="col-md-12 col-12"
        >
          <!---->
           
          <div
            class="v-card__subtitle text-left padding-0 bc-padding-left-0"
          />
           
          <div
            class="v-card__title text-left bc-padding-left-0"
          >
            
            John
            Doe
          
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-profile-email"
            rules="v => !!v || 'This field is required',v => /^\\\\w+([.-]?\\\\w+)*@\\\\w+([.-]?\\\\w+)*(\\\\.\\\\w{2,10})+$/.test(v) || 'E-mail must be valid',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <label
              class="text-left my-1"
              for="input-profile-email"
            >
              
    
    
              <!---->
            </label>
             
            <!---->
             
            <div
              class="v-input input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-profile-email"
                      id="input-profile-email"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-profile-phone"
            rules="v => !!v || 'This field is required',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <label
              class="text-left my-1"
              for="input-profile-phone"
            >
              
    
    
              <!---->
            </label>
             
            <!---->
             
            <div
              class="v-input input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-profile-phone"
                      id="input-profile-phone"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col col-12"
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <hr
              aria-orientation="horizontal"
              class="v-divider theme--light"
              role="separator"
            />
             
            <div
              class="v-card__actions"
            >
              <div
                class="spacer"
              />
               
              <button
                class="btn white--text btn-profile-update v-btn v-btn--depressed theme--light v-size--default  primary  normal"
                data-test-id="btn-profile-update"
                depressed=""
                name="button"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
                
              
                </span>
              </button>
            </div>
          </div>
        </div>
         
        <div
          class="text-center col"
        >
          <div
            class="v-dialog__container text-left overflow-y-auto"
            role="dialog"
          >
            <div
              class="v-dialog__content"
              role="document"
              style="z-index: 0;"
            >
              <div
                class="v-dialog v-dialog--persistent"
                style="width: 90%; display: none;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Profile.vue renders props when passed with gettors 1`] = `
<div
  class="mx-auto card-width v-card v-sheet theme--light"
>
  <!---->
   
  <header
    class="bc-subtitle v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <div
        class="v-toolbar__title"
      >
        
      </div>
       
      <div
        class="flex-grow-1"
      />
    </div>
  </header>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
   
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <div
      class="container"
    >
      <div
        class="row ma-5"
      >
        <div
          class="col-md-12 col-12"
        >
          <!---->
           
          <div
            class="v-card__subtitle text-left padding-0 bc-padding-left-0"
          />
           
          <div
            class="v-card__title text-left bc-padding-left-0"
          >
            
            John
            Doe
          
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-profile-email"
            rules="v => !!v || 'This field is required',v => /^\\\\w+([.-]?\\\\w+)*@\\\\w+([.-]?\\\\w+)*(\\\\.\\\\w{2,10})+$/.test(v) || 'E-mail must be valid',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <label
              class="text-left my-1"
              for="input-profile-email"
            >
              
    
    
              <!---->
            </label>
             
            <!---->
             
            <div
              class="v-input input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-profile-email"
                      id="input-profile-email"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-profile-phone"
            rules="v => !!v || 'This field is required',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <label
              class="text-left my-1"
              for="input-profile-phone"
            >
              
    
    
              <!---->
            </label>
             
            <!---->
             
            <div
              class="v-input input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-profile-phone"
                      id="input-profile-phone"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col col-12"
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <hr
              aria-orientation="horizontal"
              class="v-divider theme--light"
              role="separator"
            />
             
            <div
              class="v-card__actions"
            >
              <div
                class="spacer"
              />
               
              <button
                aria-disabled="true"
                class="btn white--text btn-profile-update v-btn v-btn--depressed v-btn--disabled theme--light v-size--default  primary  normal disabled"
                data-test-id="btn-profile-update"
                depressed=""
                disabled="disabled"
                name="button"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
                
              
                </span>
              </button>
            </div>
          </div>
        </div>
         
        <div
          class="text-center col"
        >
          <div
            class="v-dialog__container text-left overflow-y-auto"
            role="dialog"
          >
            <div
              class="v-dialog__content"
              role="document"
              style="z-index: 0;"
            >
              <div
                class="v-dialog v-dialog--persistent"
                style="width: 90%; display: none;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Profile.vue renders props when passed with gettors 2`] = `
<div
  class="mx-auto card-width v-card v-sheet theme--light"
>
  <!---->
   
  <header
    class="bc-subtitle v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <div
        class="v-toolbar__title"
      >
        
      </div>
       
      <div
        class="flex-grow-1"
      />
    </div>
  </header>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
   
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <div
      class="container"
    >
      <div
        class="row ma-5"
      >
        <div
          class="col-md-12 col-12"
        >
          <!---->
           
          <div
            class="v-card__subtitle text-left padding-0 bc-padding-left-0"
          />
           
          <div
            class="v-card__title text-left bc-padding-left-0"
          >
            
            John
            Doe
          
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-profile-email"
            rules="v => !!v || 'This field is required',v => /^\\\\w+([.-]?\\\\w+)*@\\\\w+([.-]?\\\\w+)*(\\\\.\\\\w{2,10})+$/.test(v) || 'E-mail must be valid',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <label
              class="text-left my-1"
              for="input-profile-email"
            >
              
    
    
              <!---->
            </label>
             
            <!---->
             
            <div
              class="v-input input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-profile-email"
                      id="input-profile-email"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="bc-form-text"
            data-test-id="input-profile-phone"
            rules="v => !!v || 'This field is required',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
          >
            <label
              class="text-left my-1"
              for="input-profile-phone"
            >
              
    
    
              <!---->
            </label>
             
            <!---->
             
            <div
              class="v-input input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend>
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      data-test-id="input-profile-phone"
                      id="input-profile-phone"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col col-12"
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <hr
              aria-orientation="horizontal"
              class="v-divider theme--light"
              role="separator"
            />
             
            <div
              class="v-card__actions"
            >
              <div
                class="spacer"
              />
               
              <button
                class="btn white--text btn-profile-update v-btn v-btn--depressed theme--light v-size--default  primary  normal"
                data-test-id="btn-profile-update"
                depressed=""
                name="button"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
                
              
                </span>
              </button>
            </div>
          </div>
        </div>
         
        <div
          class="text-center col"
        >
          <div
            class="v-dialog__container text-left overflow-y-auto"
            role="dialog"
          >
            <div
              class="v-dialog__content v-dialog__content--active"
              role="document"
              style="z-index: 202; z-index: 202;"
              tabindex="0"
            >
              <div
                class="v-dialog v-dialog--active v-dialog--persistent"
                style="width: 90%;"
              >
                <div
                  class="pa-8 v-card v-sheet theme--light"
                >
                  <p
                    class="text-left pl-5 ml-8"
                  >
                    
                  </p>
                   
                  <hr
                    aria-orientation="horizontal"
                    class="v-divider theme--light"
                    role="separator"
                  />
                   
                  <div
                    class="v-card__text text-left terms-content pa-0"
                    id="scroll-target"
                  >
                    <div
                      class="container container--fluid"
                    >
                      <div
                        class="row terms no-gutters"
                      >
                        <div
                          class="col-md-12 col-12"
                        >
                          <div
                            class="mx-4"
                          >
                            <h1>
                              Terms of Use
                            </h1>
                             
                            <p>
                              version 1 May 7, 2020
                            </p>
                             
                            <p>
                              
          The BC Services Card Authentication Service (the
          "
                              <strong>
                                Service
                              </strong>
                              ") allows you, the entity accepting these
          Terms of Use ("
                              <strong>
                                you
                              </strong>
                              ", "
                              <strong>
                                You
                              </strong>
                              ",
          "
                              <strong>
                                your
                              </strong>
                              " or "
                              <strong>
                                Your
                              </strong>
                              ") to access and
          use the identity and authentication services of the Provincial
          Identity Information Management Program ("
                              <strong>
                                IDIM
                              </strong>
                              ") to
          deliver your service ( "
                              <strong>
                                Your Solution
                              </strong>
                              ") to end
          users. The Service includes: (a) a test environment where you can
          ensure that Your Solution will integrate and function correctly with
          the Service; and (b) a production environment to which you may be
          given access to deploy Your Solution following an assessment by IDIM
          of Your Solution and the execution of an Information Sharing
          Agreement to address any "personal information" (as that term is
          defined in the Freedom of Information and Protection of Privacy Act
          (B.C.)) that may be exchanged between the Service and Your Solution.
        
                            </p>
                             
                            <p>
                              
          These Terms of Use are an agreement between you and Her Majesty the
          Queen in Right of the Province of British Columbia, represented by
          the Minister of Citizens' Services, IDIM (the
          "
                              <strong>
                                Province
                              </strong>
                              "), and govern your use of the Service. By
          indicating that you agree to these Terms of Use, and in
          consideration of the use of the Service, you agree to the following:
        
                            </p>
                             
                            <h2>
                              1. Authority and Ability to Accept Terms of Use
                            </h2>
                             
                            <p>
                              
          The individual accepting these Terms of Use on your behalf
          represents and warrants that: (a) s/he is at least 19 years of age;
          and (b) s/he has all necessary authority to accept these Terms of
          Use on your behalf.
        
                            </p>
                             
                            <h2>
                              2. Responsibility for Users
                            </h2>
                             
                            <p>
                              
          You acknowledge and agree that you are responsible for ensuring that
          any of your employees, agents and representatives (collectively,
          "
                              <strong>
                                Users
                              </strong>
                              ") who access or use the Service comply with
          these Terms of Use.
        
                            </p>
                             
                            <h2>
                              3. Ownership
                            </h2>
                             
                            <p>
                              
          The Service, the website through which the Service is accessed, and
          all other information and works made available, displayed or
          transmitted in connection with the Service, including without
          limitation data, text, audio, video, trademarks, trade names, logos,
          domain names, images, graphics, graphical user interface elements
          and designs, in any form or medium whatsoever (collectively, the
          "
                              <strong>
                                Content
                              </strong>
                              "), are owned by the Province or its
          licensors and are protected by copyright, patent, trademark and
          other laws protecting intellectual property rights.
        
                            </p>
                             
                            <h2>
                              4. License
                            </h2>
                             
                            <p>
                              
          Subject to your compliance with these Terms of Use, the Province
          grants you a non-exclusive, revocable, limited license to access and
          use the Service and the Content solely as necessary to test and, if
          applicable, to deploy Your Solution in accordance with these Terms
          of Use. You must not copy, distribute, republish, transmit or
          otherwise exploit any part of the Service or the Content, and the
          Province expressly reserves all intellectual property rights in and
          to the Service and the Content. If your use of the Service is
          terminated under section 5, your license is automatically
          terminated.
        
                            </p>
                             
                            <h2>
                              5. Suspension or Termination
                            </h2>
                             
                            <p>
                              
          The Province may at any time, in its sole discretion, suspend or
          terminate your use of the Service, in whole or in part: (a) if you
          fail to comply with any provision of these Terms of Use; (b) as a
          security measure; or (c) for administrative purposes.
        
                            </p>
                             
                            <h2>
                              6. Changes to Service and/or Terms of Use
                            </h2>
                             
                            <p>
                              
          The Province may at any time, in its sole discretion and without
          direct notice to you: (a) discontinue the Service; or (b) make
          changes to the Service and/or these Terms of Use. By continuing to
          use the Service, you will be conclusively deemed to have accepted
          any changes to these Terms of Use.
        
                            </p>
                             
                            <h2>
                              7. Acceptable Use
                            </h2>
                             
                            <p>
                              
          You must not take any action in connection with your use of the
          Service that would jeopardize the security, integrity and/or
          availability of the Service, including without limitation:
        
                            </p>
                             
                            <ol
                              type="a"
                            >
                              <li>
                                
            using the Service for any unlawful or inappropriate purpose;
          
                              </li>
                               
                              <li>
                                
            decompiling, disassembling, reverse engineering or otherwise
            copying any software associated with the Service;
          
                              </li>
                               
                              <li>
                                tampering with any portion of the Service;
                              </li>
                               
                              <li>
                                
            using the Service to transmit any virus or other harmful or
            destructive computer code, files or programs or to conduct hacking
            and/or intrusion activities;
          
                              </li>
                               
                              <li>
                                
            attempting to circumvent or subvert any security measure
            associated with the Service;
          
                              </li>
                               
                              <li>
                                
            taking any action that might reasonably be construed as likely to
            adversely affect other users of the Service; or
          
                              </li>
                               
                              <li>
                                
            removing or altering any proprietary symbol or notice, including
            any copyright notice, trademark or logo, displayed in connection
            with the Service.
          
                              </li>
                            </ol>
                             
                            <h2>
                              8. Privacy
                            </h2>
                             
                            <p>
                              
          When you use the Service, "contact information" (as that term is
          defined in the
          
                              <em>
                                Freedom of Information and Protection of Privacy Act
                              </em>
                              
          (B.C.)) is collected from the individual accepting these Terms of
          Use on your behalf. If you wish to be granted access to the
          production environment, you acknowledge that: (a) such access is at
          the sole discretion of the Province; (b) you will be required to
          undergo an assessment of Your Solution; and (c) you must enter into
          an Information Sharing Agreement to ensure that any exchange of
          information between the Service and Your Solution is compliant with
          applicable legislation, including without limitation the
          
                              <em>
                                Freedom of Information and Protection of Privacy Act
                              </em>
                              
          (B.C.).
        
                            </p>
                             
                            <h2>
                              9. Service Provided "As Is"
                            </h2>
                             
                            <p>
                              
          The Service is provided to you "as is", and the Province disclaims
          all representations, warranties, conditions, obligations and
          liabilities of any kind, whether express or implied, in relation to
          the Service, including without limitation implied warranties with
          respect to merchantability, satisfactory quality, fitness for a
          particular purpose and non-infringement. Without limiting the
          general nature of the previous sentence, the Province does not
          represent or warrant that:
        
                            </p>
                             
                            <ol
                              type="a"
                            >
                              <li>
                                the Service will be available;
                              </li>
                               
                              <li>
                                
            your use of the Service will be timely, uninterrupted or error
            free;
          
                              </li>
                               
                              <li>
                                any errors in the Service will be corrected; or
                              </li>
                               
                              <li>
                                the Service will meet your expectations and requirements.
                              </li>
                            </ol>
                             
                            <h2>
                              10. No Liability
                            </h2>
                             
                            <p>
                              
          To the maximum extent permitted by applicable law, under no
          circumstances will the Province be liable to any person or entity
          for any direct, indirect, special, incidental, consequential or
          other loss, claim, injury or damage, whether foreseeable or
          unforeseeable (including without limitation claims for damages for
          loss of profits or business opportunities, use or misuse of, or
          inability to use, the Service, interruptions, deletion or corruption
          of files, loss of programs or information, errors, defects or
          delays) arising out of or in any way connected with your use of the
          Service and whether based on contract, tort, strict liability or any
          other legal theory. The previous sentence will apply even if the
          Province has been specifically advised of the possibility of any
          such loss, claim, injury or damage.
        
                            </p>
                             
                            <h2>
                              11. Indemnity
                            </h2>
                             
                            <p>
                              
          You agree to indemnify, defend and hold harmless the Province and
          the Province's employees and agents from and against all claims,
          demands, obligations, losses, liabilities, costs or debts, and
          expenses (including but not limited to reasonable legal fees)
          arising from: (a) your, and your Users', use of the Service; or (b)
          your, or any User's, violation of any provision of these Terms of
          Use.
        
                            </p>
                             
                            <h2>
                              12. General
                            </h2>
                             
                            <p>
                              
          These Terms of Use are the entire agreement between you and the
          Province with respect to the subject matter of these Terms of Use.
          The headings in these Terms of Use are inserted for convenience only
          and will not be used in interpreting or construing any provision of
          these Terms of Use. If any provision of these Terms of Use is
          invalid, illegal or unenforceable, that provision will be severed
          from these Terms of Use and all other provisions will remain in full
          force and effect. These Terms of Use will be governed by and
          construed in accordance with the laws of the province of British
          Columbia and the applicable laws of Canada. By using the Service,
          you consent to the exclusive jurisdiction and venue of the courts of
          the province of British Columbia, sitting in Victoria, for the
          hearing of any dispute arising from or related to these Terms of Use
          and their subject matter.
        
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                     
                    <div />
                  </div>
                   
                  <hr
                    aria-orientation="horizontal"
                    class="v-divider theme--light"
                    role="separator"
                  />
                   
                  <div
                    class="v-card__actions terms-actions"
                  >
                    <div
                      class="spacer"
                    />
                     
                    <button
                      aria-label="Back Button"
                      class="btn v-btn v-btn--contained theme--light v-size--default  secondary-btn  normal"
                      data-test-id="btn-cancel-terms-profile"
                      name="button"
                      type="button"
                    >
                      <span
                        class="v-btn__content"
                      >
                        
                      </span>
                    </button>
                     
                    <button
                      class="btn white--text btn-submit-terms-profile ml-6 v-btn v-btn--depressed theme--light v-size--default  primary  normal"
                      data-test-id="btn-submit-terms-profile"
                      depressed=""
                      name="button"
                      type="button"
                    >
                      <span
                        class="v-btn__content"
                      >
                        
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
